// models/organization_pb.js
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.com.chicago.dto.Organization', null, global);
goog.exportSymbol('proto.com.chicago.dto.OrganizationInfo', null, global);
goog.exportSymbol('proto.com.chicago.dto.OrganizationType', null, global);

/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.chicago.dto.Organization = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.chicago.dto.Organization.repeatedFields_, null);
};
goog.inherits(proto.com.chicago.dto.Organization, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  proto.com.chicago.dto.Organization.displayName = 'proto.com.chicago.dto.Organization';
}
/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.chicago.dto.Organization.repeatedFields_ = [9,10,11,12];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto suitable for use in Soy templates.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     com.google.apps.jspb.JsClassTemplate.JS_RESERVED_WORDS.
 * @param {boolean=} opt_includeInstance Whether to include the JSPB instance
 *     for transitional soy proto support: http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.chicago.dto.Organization.prototype.toObject = function(opt_includeInstance) {
  return proto.com.chicago.dto.Organization.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Whether to include the JSPB
 *     instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.chicago.dto.Organization} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.chicago.dto.Organization.toObject = function(includeInstance, msg) {
  var f, obj = {
    type: jspb.Message.getFieldWithDefault(msg, 1, 0),
    organizationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    name: jspb.Message.getFieldWithDefault(msg, 3, ""),
    description: jspb.Message.getFieldWithDefault(msg, 4, ""),
    webSite: jspb.Message.getFieldWithDefault(msg, 5, ""),
    emailDomain: jspb.Message.getFieldWithDefault(msg, 6, ""),
    parentOrganizationId: jspb.Message.getFieldWithDefault(msg, 7, ""),
    entityId: jspb.Message.getFieldWithDefault(msg, 8, ""),
    phonesList: jspb.Message.getRepeatedField(msg, 9),
    faxList: jspb.Message.getRepeatedField(msg, 10),
    usersList: jspb.Message.getRepeatedField(msg, 11),
    organizationsList: jspb.Message.getRepeatedField(msg, 12),
    addressId: jspb.Message.getFieldWithDefault(msg, 13, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.chicago.dto.Organization}
 */
proto.com.chicago.dto.Organization.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.chicago.dto.Organization;
  return proto.com.chicago.dto.Organization.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.chicago.dto.Organization} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.chicago.dto.Organization}
 */
proto.com.chicago.dto.Organization.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.com.chicago.dto.OrganizationType} */ (reader.readEnum());
      msg.setType(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganizationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setDescription(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setWebSite(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setEmailDomain(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setParentOrganizationId(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setEntityId(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.addPhones(value);
      break;
    case 10:
      var value = /** @type {string} */ (reader.readString());
      msg.addFax(value);
      break;
    case 11:
      var value = /** @type {string} */ (reader.readString());
      msg.addUsers(value);
      break;
    case 12:
      var value = /** @type {string} */ (reader.readString());
      msg.addOrganizations(value);
      break;
    case 13:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddressId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.chicago.dto.Organization.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.chicago.dto.Organization.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.chicago.dto.Organization} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.chicago.dto.Organization.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getType();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getOrganizationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getDescription();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getWebSite();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getEmailDomain();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getParentOrganizationId();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getEntityId();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
  f = message.getPhonesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      9,
      f
    );
  }
  f = message.getFaxList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      10,
      f
    );
  }
  f = message.getUsersList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      11,
      f
    );
  }
  f = message.getOrganizationsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      12,
      f
    );
  }
  f = message.getAddressId();
  if (f.length > 0) {
    writer.writeString(
      13,
      f
    );
  }
};


/**
 * optional OrganizationType type = 1;
 * @return {!proto.com.chicago.dto.OrganizationType}
 */
proto.com.chicago.dto.Organization.prototype.getType = function() {
  return /** @type {!proto.com.chicago.dto.OrganizationType} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/** @param {!proto.com.chicago.dto.OrganizationType} value */
proto.com.chicago.dto.Organization.prototype.setType = function(value) {
  jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional string organization_id = 2;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getOrganizationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setOrganizationId = function(value) {
  jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string name = 3;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setName = function(value) {
  jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string description = 4;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setDescription = function(value) {
  jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string web_site = 5;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getWebSite = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setWebSite = function(value) {
  jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string email_domain = 6;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getEmailDomain = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setEmailDomain = function(value) {
  jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string parent_organization_id = 7;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getParentOrganizationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setParentOrganizationId = function(value) {
  jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional string entity_id = 8;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getEntityId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setEntityId = function(value) {
  jspb.Message.setProto3StringField(this, 8, value);
};


/**
 * repeated string phones = 9;
 * @return {!Array.<string>}
 */
proto.com.chicago.dto.Organization.prototype.getPhonesList = function() {
  return /** @type {!Array.<string>} */ (jspb.Message.getRepeatedField(this, 9));
};


/** @param {!Array.<string>} value */
proto.com.chicago.dto.Organization.prototype.setPhonesList = function(value) {
  jspb.Message.setField(this, 9, value || []);
};


/**
 * @param {!string} value
 * @param {number=} opt_index
 */
proto.com.chicago.dto.Organization.prototype.addPhones = function(value, opt_index) {
  jspb.Message.addToRepeatedField(this, 9, value, opt_index);
};


proto.com.chicago.dto.Organization.prototype.clearPhonesList = function() {
  this.setPhonesList([]);
};


/**
 * repeated string fax = 10;
 * @return {!Array.<string>}
 */
proto.com.chicago.dto.Organization.prototype.getFaxList = function() {
  return /** @type {!Array.<string>} */ (jspb.Message.getRepeatedField(this, 10));
};


/** @param {!Array.<string>} value */
proto.com.chicago.dto.Organization.prototype.setFaxList = function(value) {
  jspb.Message.setField(this, 10, value || []);
};


/**
 * @param {!string} value
 * @param {number=} opt_index
 */
proto.com.chicago.dto.Organization.prototype.addFax = function(value, opt_index) {
  jspb.Message.addToRepeatedField(this, 10, value, opt_index);
};


proto.com.chicago.dto.Organization.prototype.clearFaxList = function() {
  this.setFaxList([]);
};


/**
 * repeated string users = 11;
 * @return {!Array.<string>}
 */
proto.com.chicago.dto.Organization.prototype.getUsersList = function() {
  return /** @type {!Array.<string>} */ (jspb.Message.getRepeatedField(this, 11));
};


/** @param {!Array.<string>} value */
proto.com.chicago.dto.Organization.prototype.setUsersList = function(value) {
  jspb.Message.setField(this, 11, value || []);
};


/**
 * @param {!string} value
 * @param {number=} opt_index
 */
proto.com.chicago.dto.Organization.prototype.addUsers = function(value, opt_index) {
  jspb.Message.addToRepeatedField(this, 11, value, opt_index);
};


proto.com.chicago.dto.Organization.prototype.clearUsersList = function() {
  this.setUsersList([]);
};


/**
 * repeated string organizations = 12;
 * @return {!Array.<string>}
 */
proto.com.chicago.dto.Organization.prototype.getOrganizationsList = function() {
  return /** @type {!Array.<string>} */ (jspb.Message.getRepeatedField(this, 12));
};


/** @param {!Array.<string>} value */
proto.com.chicago.dto.Organization.prototype.setOrganizationsList = function(value) {
  jspb.Message.setField(this, 12, value || []);
};


/**
 * @param {!string} value
 * @param {number=} opt_index
 */
proto.com.chicago.dto.Organization.prototype.addOrganizations = function(value, opt_index) {
  jspb.Message.addToRepeatedField(this, 12, value, opt_index);
};


proto.com.chicago.dto.Organization.prototype.clearOrganizationsList = function() {
  this.setOrganizationsList([]);
};


/**
 * optional string address_id = 13;
 * @return {string}
 */
proto.com.chicago.dto.Organization.prototype.getAddressId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 13, ""));
};


/** @param {string} value */
proto.com.chicago.dto.Organization.prototype.setAddressId = function(value) {
  jspb.Message.setProto3StringField(this, 13, value);
};



/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.chicago.dto.OrganizationInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.chicago.dto.OrganizationInfo.repeatedFields_, null);
};
goog.inherits(proto.com.chicago.dto.OrganizationInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  proto.com.chicago.dto.OrganizationInfo.displayName = 'proto.com.chicago.dto.OrganizationInfo';
}
/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.chicago.dto.OrganizationInfo.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto suitable for use in Soy templates.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     com.google.apps.jspb.JsClassTemplate.JS_RESERVED_WORDS.
 * @param {boolean=} opt_includeInstance Whether to include the JSPB instance
 *     for transitional soy proto support: http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.com.chicago.dto.OrganizationInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Whether to include the JSPB
 *     instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.chicago.dto.OrganizationInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.chicago.dto.OrganizationInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    type: jspb.Message.getFieldWithDefault(msg, 1, 0),
    organizationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    name: jspb.Message.getFieldWithDefault(msg, 3, ""),
    organizationsList: jspb.Message.toObjectList(msg.getOrganizationsList(),
    proto.com.chicago.dto.OrganizationInfo.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.chicago.dto.OrganizationInfo}
 */
proto.com.chicago.dto.OrganizationInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.chicago.dto.OrganizationInfo;
  return proto.com.chicago.dto.OrganizationInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.chicago.dto.OrganizationInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.chicago.dto.OrganizationInfo}
 */
proto.com.chicago.dto.OrganizationInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.com.chicago.dto.OrganizationType} */ (reader.readEnum());
      msg.setType(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganizationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 4:
      var value = new proto.com.chicago.dto.OrganizationInfo;
      reader.readMessage(value,proto.com.chicago.dto.OrganizationInfo.deserializeBinaryFromReader);
      msg.addOrganizations(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.chicago.dto.OrganizationInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.chicago.dto.OrganizationInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.chicago.dto.OrganizationInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getType();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getOrganizationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getOrganizationsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.com.chicago.dto.OrganizationInfo.serializeBinaryToWriter
    );
  }
};


/**
 * optional OrganizationType type = 1;
 * @return {!proto.com.chicago.dto.OrganizationType}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.getType = function() {
  return /** @type {!proto.com.chicago.dto.OrganizationType} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/** @param {!proto.com.chicago.dto.OrganizationType} value */
proto.com.chicago.dto.OrganizationInfo.prototype.setType = function(value) {
  jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional string organization_id = 2;
 * @return {string}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.getOrganizationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/** @param {string} value */
proto.com.chicago.dto.OrganizationInfo.prototype.setOrganizationId = function(value) {
  jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string name = 3;
 * @return {string}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/** @param {string} value */
proto.com.chicago.dto.OrganizationInfo.prototype.setName = function(value) {
  jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * repeated OrganizationInfo organizations = 4;
 * @return {!Array.<!proto.com.chicago.dto.OrganizationInfo>}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.getOrganizationsList = function() {
  return /** @type{!Array.<!proto.com.chicago.dto.OrganizationInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.chicago.dto.OrganizationInfo, 4));
};


/** @param {!Array.<!proto.com.chicago.dto.OrganizationInfo>} value */
proto.com.chicago.dto.OrganizationInfo.prototype.setOrganizationsList = function(value) {
  jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.com.chicago.dto.OrganizationInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.chicago.dto.OrganizationInfo}
 */
proto.com.chicago.dto.OrganizationInfo.prototype.addOrganizations = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.com.chicago.dto.OrganizationInfo, opt_index);
};


proto.com.chicago.dto.OrganizationInfo.prototype.clearOrganizationsList = function() {
  this.setOrganizationsList([]);
};


/**
 * @enum {number}
 */
proto.com.chicago.dto.OrganizationType = {
  HOLDING: 0,
  COMPANY: 1,
  BRANCH: 2
};

goog.object.extend(exports, proto.com.chicago.dto);
